src/assembler/instruction.h

```diff
@@ -103,14 +103,15 @@ struct Opcode : public Field {
 };
 
 struct Reg : public Field {
+    enum class Regsd { rd, rs1, rs2 };
```

Comment:
General comment: In the spirit of the ISA-agnostic way that the assembler is written, it might be better for the register "names" to simply be defined as strings.